src/vips/Image_methods.lua

```diff
@@ -9,6 +9,14 @@ local vobject = require "vips.vobject"
 local voperation = require "vips.voperation"
 local Image = require "vips.Image"
 
+local type = type
```

Comment:
I've not seen this -- could you explain ow local helps speed here? Are they just faster than globals?

Reply:
>Local variables are very fast as they reside in virtual machine registers, and are accessed directly by index. Global variables on the other hand, reside in a Lua table and as such are accessed by a hash lookup.

Indeed, this is weird, and it seems that this is common. Every `lua-resty-*` package that I checked, seems to have this:
[lua-resty-core](https://github.com/openresty/lua-resty-core/blob/bdbac701eb017370775f7333979922041021aeee/lib/resty/core/hash.lua#L9)
[lua-resty-http](https://github.com/pintsized/lua-resty-http/blob/fe5c10a47cf40440845c140a5d29cd0e0cd0208f/lib/resty/http.lua#L40)
[lua-resty-redis](https://github.com/openresty/lua-resty-redis/blob/41827e454126d8a030b9e09fbf7e3a9f3afb8088/lib/resty/redis.lua#L8)
[lua-resty-template](https://github.com/bungle/lua-resty-template/blob/214bbe3e5cdd60ed7c321c0f99579c519ef346e1/lib/resty/template.lua#L16)

Let me know if I should revert this.

Reply:
No that's fine, it just looks strange. Good to know!